### DISCUSSION TITLE
Replace (most of) DOMString/ByteString/USVString/CSSOMString with just String?

### DISCUSSION THREAD

**Discussion by foolip** [mdn/mdn-community#541] [Content]
[`DOMString`](https://heycam.github.io/webidl/#idl-DOMString), [`ByteString`](https://heycam.github.io/webidl/#idl-ByteString) and [`USVString`](https://heycam.github.io/webidl/#idl-USVString) are defined by Web IDL as strings with varying restrictions, while [`CSSOMString`](https://drafts.csswg.org/cssom/#cssomstring-type) is defined as either `DOMString` or `USVString`, to allow for implementations that use either UTF-8 or UTF-16 internally.

From the perspective of JavaScript and web developers, there are just strings:

`DOMString` is just a `String` by another name.

`ByteString` has the additional requirement that no code point is larger than 255 and is used mainly for HTTP headers. The following throws a `TypeError` because of this:
```js
const h = new Headers();
h.append('something', '\u0100'); // "String contains non ISO-8859-1 code point" error in Chrome
```

 `USVString` has the additional requirement that there are no lone surrogates, and is used a lot for URLs. Lone surrogates are replaced with U+FFFD in APIs that accept USVString.

`CSSOMString` is just a type alias, so it might behave like `USVString`, or it might not.

These distinctions rarely matter, and **I'm proposing that MDN simply talks about strings in most cases**.

Aside: There's an unfortunate number of references to "DOMString objects", "USVString object", etc. There aren't objects, they're strings :)

**Comment by sideshowbarker** [comment]
> Replace (most of) DOMString/ByteString/USVString/CSSOMString with just String?
> These distinctions rarely matter, and **I'm proposing that MDN simply talks about strings in most cases**.

Big +1 👍 I strongly support making this change.

**Reply by teoli2003** [reply]
+1 too! I even opened #5614 a few weeks ago.

(For ByteString we should make sure the TypeError is listed and explained)

**Reply by foolip** [reply]
Awesome, I'm glad I'm not the only one who gotten hung up on this :)

**Comment by pepelsbey** [comment]
@foolip is it still relevant? There are many `DOMString` in content, but only a few `ByteString`, `USVString`, and `CSSOMString`.
